### PR TITLE
Remove icon wrapper

### PR DESCRIPTION
**Pull request by hututudor**
https://etvas.atlassian.net/browse/ETV-4296

### CODE REVIEW

File: src/Button/Button.jsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ import { Icon } from '../Icon'
 import sizes from '../assets/sizes'
 
 const Button = ({
+  color,
```
Comment (sergiuelisei) [inline src/Button/Button.jsx:14]: Please use the color prop for the right icon too

**Review by sergiuelisei — CHANGES_REQUESTED** [review]

**Review by sergiuelisei — APPROVED** [review]

File: src/Icon/Icon.jsx [hunk 2]
```diff
@@ -31,24 +31,21 @@ const validate = name => {
...
-      color='currentColor'
-      spin={rotate}
-      {...props}
-    />
-  </Wrapper>
+  <BaseIcon
+    path={externalGlyphs[name] || glyphs[name] || validate(name)}
+    size={sizes[size] ?? size}
+    color={color}
+    spin={rotate}
```
Comment (alsofronie) [inline src/Icon/Icon.jsx:38]:
```suggestion
    spin={spin}
    rotate={typeof rotate === 'string' ? rotate : `${rotate}deg`}
```

File: src/Button/Button.jsx [hunk 3]
```diff
@@ -42,15 +43,15 @@ const Button = ({
       hSpacing={hSpacing}
       {...rest}>
       {loading ? (
-        <Icon size='medium' name='loading' rotate={true} />
+        <Icon size='medium' name='loading' spin={true} color={iconColor} />
```
Comment (alsofronie) [inline src/Button/Button.jsx:46]: Can't this just be:

```suggestion
        <Icon size='medium' name='loading' spin color={iconColor} />
```

File: src/Icon/Icon.jsx [hunk 4]
```diff
@@ -68,7 +66,8 @@ Icon.propTypes = {
     PropTypes.object,
     PropTypes.string
   ]),
-  rotate: PropTypes.bool
+  rotate: PropTypes.number,
```
Comment (alsofronie) [inline src/Icon/Icon.jsx:69]:
```suggestion
  rotate: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
```

File: src/Icon/Icon.jsx [hunk 5]
```diff
@@ -30,25 +30,23 @@ const validate = name => {
...
-      {...props}
-    />
-  </Wrapper>
+export const Icon = ({ name, size, color, rotate, spin, ...props }) => (
+  <BaseIcon
+    path={externalGlyphs[name] || glyphs[name] || validate(name)}
+    size={sizes[size] ?? size}
+    color={color}
+    spin={spin}
+    rotate={rotate}
```
Comment (alsofronie) [inline src/Icon/Icon.jsx:39]: Is the string/number distinction handled by the BaseIcon? Or we need to do:

```suggestion
    rotate={typeof rotate === 'string' ? rotate : `${rotate}deg`}
```

Comment (hututudor) [reply src/Icon/Icon.jsx:39]: The BaseIcon accepts only numbers and computes the string itself

We do not need to accept both string and numbers in our component

**Review by alsofronie — APPROVED** [review]

**Pull request merged**